src/compiler/checker.ts

```diff
@@ -15798,7 +15798,7 @@ namespace ts {
         function substituteIndexedMappedType(objectType: MappedType, index: Type) {
             const mapper = createTypeMapper([getTypeParameterFromMappedType(objectType)], [index]);
             const templateMapper = combineTypeMappers(objectType.mapper, mapper);
-            return instantiateType(getTemplateTypeFromMappedType(objectType), templateMapper);
+            return instantiateType(getTemplateTypeFromMappedType(objectType.target as MappedType || objectType), templateMapper);

```

Comment:
Is `.target` like a "link" to the original type if you end up creating a new instantiated type from something? 🤔

Reply:
Yes, an instantiated type has a `target` property that links to the original mapped type and a `mapper` property that provides a mapping from type parameters to corresponding type arguments. The original mapped type doesn't have `target` and `mapper` properties. It's target is itself and it's type parameters are also its type arguments, so to speak.